### ISSUE TITLE
Environment variable to control Cassandra TTL options

### ISSUE THREAD

**Issue by mikewrighton** [openzipkin/zipkin#1053]
It would be useful if there was an environment variable to control the span and index TTL options for Cassandra (zipkin.store.cassandra.spanTTL). e.g. `CASSANDRA_SPAN_TTL`.

**Comment by eirslett** [comment]
Maybe we should do this more generally, e.g. all zipkin flags should be able to read from environment variables with the same name; e.g. the "zipkin.collector.storage.ttl" should be able to read the environment variable ZIPKIN_COLLECTOR_STORAGE_TTL. That should make it work for all flags - a more generic solution. @adriancole wdyt? Would this require changes in twitter libraries upstream?

**Comment by codefromthecrypt** [comment]
Right now, it is nice to know what the hardest hitting properties are, ex that they've been asked for. There are a number of internal properties that have never been modified, and I really don't want to expand the interface or make portability more difficult by locking them in.

**Comment by codefromthecrypt** [comment]
To elaborate :)

I'd like to separate functionality from support policy. Whether or not some library inside twitter.app automagically accepts env variables is a different point than the list of known tested properties.

By supporting a closed set of env properties, backends can work cross-language. Ex. someone writing a collector in go has a limited set of well-known properties they'd need to support. This works today. For example, the closed set of env variables is why you can drop in the java or the scala query server and get the same behavior in docker (service-level abstraction). All of the readmes include a closed set of properties and what they do. I've found a closed set quite helpful when discussing support issues with @prat0318 for example.

On the functionality side, Spring boot does auto-export, but I consider that an internal detail. Ex Some of the names aren't well thought through, or eagerly configurable. If we formally support implicitly defined variables, we'd have a harder time removing or changing them in the future. Sometimes changing them is needed to better design the component. For example, some things keep an internal queue when maybe they shouldn't.

Long story short, if/when finatra could support auto-export, and we could use that feature, I would still want to explicitly call out TTL as a supported feature vs an accident of a flag name. I do find value in such a feature as it could allow us to experiment with not-yet-supported flags easier.

Hope this essay helps :)

PS #1041 would need to be addressed before a flag auto-env-binder would be usable.
